tests/RobotFramework/tests/mqtt/local_broker_authentication/local_broker_authentication.robot

```diff
@@ -0,0 +1,131 @@
+
+Username/password authentication to local broker with TLS - mosquitto bridge
+    [Setup]    TLS Setup    use_builtin_bridge=false
+    Check system health including c8y
+
+Username/password authentication to local broker with TLS - built-in bridge
+    [Setup]    TLS Setup    use_builtin_bridge=true
+    Check system health including c8y
+
+
+*** Keywords ***
+Check system health including c8y
+    ThinEdgeIO.Set Device Context    ${CONTAINER_1}
+    ThinEdgeIO.Service Should Be Running    mosquitto
+    ThinEdgeIO.Service Should Be Running    tedge-mapper-c8y
+    ThinEdgeIO.Service Should Be Running    tedge-agent
+    ThinEdgeIO.Service Should Be Running    c8y-firmware-plugin
+
+    ThinEdgeIO.Set Device Context    ${CONTAINER_2}
+    ThinEdgeIO.Service Should Be Stopped    mosquitto
+    ThinEdgeIO.Service Should Be Stopped    tedge-mapper-c8y
+    ThinEdgeIO.Service Should Be Running    tedge-agent
+    ThinEdgeIO.Service Should Be Stopped    c8y-firmware-plugin
+
+    # tedge mqtt pub/sub sanity check
+    ${output}=    Execute Command    tedge mqtt pub test-topic test-message    stderr=${True}
+    Should Not Contain    ${output}    Failed to connect to broker
+    ${output}=    Execute Command    tedge mqtt sub test --duration 1s    stderr=${True}
+    Should Not Contain    ${output}    Failed to connect to broker
+
+    # Validate the device exists in the cloud
+    Cumulocity.Device Should Exist    ${CONTAINER_1}
+
+    # Cumulocity sanity check
+    ThinEdgeIO.Set Device Context    ${CONTAINER_2}
+    ThinEdgeIO.Execute Command    tedge mqtt pub te/device/main///m/ '{"temperature": 29.8}'
+    ${measurements}=    Cumulocity.Device Should Have Measurements    value=temperature    series=temperature
+    Should Be Equal As Numbers    ${measurements[0]["temperature"]["temperature"]["value"]}    ${29.8}
+
+    Cumulocity.Should Have Services    name=tedge-mapper-c8y    status=up
+    Cumulocity.Should Have Services    name=tedge-agent    status=up
+    Cumulocity.Should Have Services    name=c8y-firmware-plugin    status=up
+
+No TLS Setup
+    [Arguments]    ${use_builtin_bridge}
+
+    # Parent
+    ${CONTAINER_1}=    Setup    bootstrap_args=--no-secure    register=${True}    connect=${False}
+    Set Test Variable    $CONTAINER_1
+    ${CONTAINER_1_HOSTNAME}=    Execute Command    hostname    strip=${True}
+
+    Execute Command    sudo tedge config set mqtt.bridge.built_in ${use_builtin_bridge}
+    Execute Command    mosquitto_passwd -c -b /etc/mosquitto/pwfile testuser testpassword
+
+    Transfer To Device    ${CURDIR}/unencrypted-listener.conf    /etc/mosquitto/conf.d/
+    Configure MQTT client    host=127.0.0.1    port=1884
+
+    Connect Mapper    c8y
+
+    # Child
+    ${CONTAINER_2}=    Setup    bootstrap_args=--no-secure    register=${False}    connect=${False}
+    Set Test Variable    $CONTAINER_2
+
+    Configure MQTT client    host=${CONTAINER_1_HOSTNAME}    port=1884
+    Start Service    tedge-agent
+
+TLS Setup
```

Comment:
Optional suggestion: Instead of two variants `TLS Setup` and `No TLS Setup`, why not just keep a single `Setup` keyword that takes `tld_enabled` as an argument? Especially since these keywords are already taking the `use_builtin_bridge` argument?

Reply:
`use_builtin_bridge` makes really one-liner difference (`sudo tedge config set mqtt.bridge.built_in ${use_builtin_bridge}`), while TLS/no TLS need more different setup (bootstrap_args, port, set/unset some tedge config keys, different listener file). Then eventually I thought it would be easier to read, if I prepare two different Setup keys.